src/pages/Lend/LoanSearchPage.vue

```diff
@@ -241,7 +241,14 @@ export default {
 				this.challengeData = goalsData?.goals?.values?.[0] || {};
 				const teamData = this.apollo.readQuery({ query: TeamInfoFromId, variables: { team_id: teamId } });
 				this.teamData = teamData?.community?.team || {};
-				setChallengeCookieData(this.cookieStore, teamId, this.teamData?.name);
+
+				const loanIds = this.challengeData?.targets?.values?.map(target => target?.loanId) ?? [];
```

Comment:
The challenge doesn't just apply to the recommended loans. It applies to any loan added from `/lend/filter`. So we need to add/update this cookie value whenever the user adds to basket from `/lend/filter` when the challenge is active on that page for the current user.